File: app/src/main/java/com/security/passwordmanager/data/MainDatabase.java
package com.security.passwordmanager.data;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.migration.Migration;
import androidx.sqlite.db.SupportSQLiteDatabase;

import com.security.passwordmanager.settings.SettingsDao;
import com.security.passwordmanager.settings.Support;

import kotlin.jvm.Volatile;

@Database(entities = {Website.class, BankCard.class, Support.Settings.class}, version = 2, exportSchema = false)
public abstract class MainDatabase extends RoomDatabase {

    @Volatile
    private static MainDatabase INSTANCE = null;

    private static final Migration MIGRATION_1_2 = new Migration(1, 2) {
        @Override
        public void migrate(@NonNull SupportSQLiteDatabase database) {
            database.execSQL("CREATE TABLE SettingsTable (id INTEGER PRIMARY KEY NOT NULL, theme TEXT)");
        }
    };

    public static MainDatabase getDatabase(Context context) {
        MainDatabase temp = INSTANCE;

        if (temp != null)
            return temp;

        synchronized (MainDatabase.class) {
            MainDatabase instance = Room
                    .databaseBuilder(context, MainDatabase.class, "PasswordBase")
                    .addMigrations(MIGRATION_1_2)
                    .allowMainThreadQueries()
                    .build();

            INSTANCE = instance;
            return instance;
        }
    }

    public abstract DataDao websiteDao();

    public abstract SettingsDao settingsDao();
}

File: app/src/main/java/com/security/passwordmanager/data/MainDatabase.kt
package com.security.passwordmanager.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import androidx.room.migration.Migration
import com.security.passwordmanager.model.BankCard
import com.security.passwordmanager.model.Settings
import com.security.passwordmanager.model.Website
import com.security.passwordmanager.settings.SettingsDao

@Database(entities = [Website::class, BankCard::class, Settings::class], version = 4, exportSchema = false)
abstract class MainDatabase : RoomDatabase() {

    abstract fun dataDao() : DataDao

    abstract fun settingsDao() : SettingsDao

    companion object {
        @Volatile
        private var INSTANCE: MainDatabase? = null

        private val MIGRATION_1_2 = Migration(1, 2) {
            for (property in arrayOf("isUsingBeautifulFont", "isShowingDataHints", "isUsingBottomView"))
                it.execSQL(
                    "ALTER TABLE SettingsTable ADD COLUMN $property INTEGER DEFAULT 0 NOT NULL")
        }

        private val MIGRATION_2_3 = Migration(2, 3) {
            it.execSQL("ALTER TABLE SettingsTable ADD COLUMN email TEXT NOT NULL DEFAULT ''")
            it.execSQL("ALTER TABLE WebsiteTable ADD COLUMN email TEXT NOT NULL DEFAULT ''")
            it.execSQL("ALTER TABLE BankTable ADD COLUMN email TEXT NOT NULL DEFAULT ''")
        }

        private val MIGRATION_3_4 = Migration(3, 4) {
            it.execSQL("ALTER TABLE BankTable ADD COLUMN bankCardName TEXT NOT NULL DEFAULT ''")
            it.execSQL("ALTER TABLE BankTable ADD COLUMN comment TEXT NOT NULL DEFAULT ''")
        }

        fun getDatabase(context : Context) : MainDatabase {
            val tempInstance = INSTANCE

            if (tempInstance != null)
                return tempInstance

            synchronized(this) {
                val instance = Room.databaseBuilder(
                        context.applicationContext,
                        MainDatabase::class.java,
                        "PasswordBase"
                )
                    .allowMainThreadQueries()
                    .addMigrations(MIGRATION_1_2, MIGRATION_2_3, MIGRATION_3_4)
                    .build()

                INSTANCE = instance
                return instance
            }
        }
    }
}